plugins/microsoft_intune/icon_microsoft_intune/util/api.py

```diff
@@ -16,6 +16,96 @@ def __init__(self, username, password, client_id, client_secret, tenant_id, api_
+
+        if not policy_id:
+            raise PluginException(
+                cause=f"Policy: {policy_name}, was not found.",
+                assistance="Contact support for help. See log for more details.",
+            )
+
+        managed_app_policies_with_apps = self._call_api(
+            "GET", f"deviceAppManagement/{device_type}ManagedAppProtections('{policy_id}')?$expand=apps"
+        )
+        managed_app_list = self._call_api("GET", "deviceAppManagement/managedAppStatuses('managedAppList')")
+        application_package_id = self._filter_app_package_id(managed_app_list, application_name)
+
+        if not application_package_id:
+            raise PluginException(
+                cause=f"Application: {application_name}, was not found.",
+                assistance="Contact support for help. See log for more details.",
+            )
+
+        target_apps = []
+
+        for item in managed_app_policies_with_apps["apps"]:
+            target_apps.append(
+                {
+                    "mobileAppIdentifier": {
+                        "@odata.type": item["mobileAppIdentifier"]["@odata.type"],
+                        "packageId": item["mobileAppIdentifier"]["packageId"],
+                    }
+                }
+            )
+
+        target_apps.append(
+            {
+                "mobileAppIdentifier": {
+                    "@odata.type": f"#microsoft.graph.{device_type}MobileAppIdentifier",
+                    "packageId": application_package_id,
+                }
+            }
+        )
+
+        return self._call_api(
+            "POST",
+            f"deviceAppManagement/{device_type}ManagedAppProtections('{policy_id}')/targetApps",
+            request_body={"apps": target_apps},
+        )
+
+    def delete_app_from_policy(self, application_name: str, policy_name: str, device_type: str):
+        managed_app_policies = self._call_api("GET", "/deviceAppManagement/managedAppPolicies")
+        policy_id = self._filter_policy_id(managed_app_policies, policy_name)
+
+        if not policy_id:
+            raise PluginException(
+                cause=f"Policy: {policy_name}, was not found.",
+                assistance="Contact support for help. See log for more details.",
+            )
+
+        managed_app_policies_with_apps = self._call_api(
+            "GET", f"deviceAppManagement/{device_type}ManagedAppProtections('{policy_id}')?$expand=apps"
+        )
+        managed_app_list = self._call_api("GET", "deviceAppManagement/managedAppStatuses('managedAppList')")
+        application_package_id = self._filter_app_package_id(managed_app_list, application_name)
+
+        if not application_package_id:
+            raise PluginException(
+                cause=f"Application: {application_name}, was not found.",
+                assistance="Contact support for help. See log for more details.",
+            )
+
+        target_apps = []
+
+        for item in managed_app_policies_with_apps["apps"]:
```

Comment:
```suggestion
        for item in managed_app_policies_with_apps.get("apps"):
```
Prevents `KeyError` if "apps" not in `managed_app_policies_with_apps`